explore/src/main/scala/explore/attachments/ObsAttachmentsTable.scala

```diff
@@ -209,9 +233,27 @@ object ObsAttachmentsTable extends TableHooks:
       .withHooks[Props]
       .useContext(AppContext.ctx)
       .useStateView(Action.None)
+      .useStateView[UrlMap](Map.empty)
+      .useEffectWithDepsBy((props, _, _, _) => props.obsAttachments.get)((props, _, _, urlMap) =>
+        obsAttachments =>
+          val newOas         = obsAttachments.filter(oa => !urlMap.get.contains(oa.toMapKey))
+          val allCurrentKeys = obsAttachments.map(_.toMapKey).toSet
+
+          val updateUrlMap =
+            urlMap
+              .mod { umap =>
+                val filteredMap = umap.filter((k, v) => allCurrentKeys.contains(k))
+                newOas.foldRight(filteredMap)((oa, m) => m.updated(oa.toMapKey, Pot.pending))
+              }
+              .to[IO]
+          val getUrls      =
+            newOas.traverse_(oa => getAttachmentUrl(props, oa, urlMap))
+
+          updateUrlMap *> getUrls
+      )
       // Columns
-      .useMemoBy((_, _, _) => ())((props, ctx, action) =>
-        _ =>
+      .useMemoBy((_, _, _, urlMap) => urlMap.reuseByValue)((props, ctx, action, _) =>
```

Comment:
You can use `urlMap.get` in the dependencies, since only the value seems to be used.

Reply:
Good point

Reply:
Good point